build_tools/cmake/iree_cross_compile.cmake

```diff
@@ -0,0 +1,226 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(iree_macros)
+
+# iree_create_configuration
+#
+# Creates custom commands and targets for an IREE configuration. An IREE
+# configuration means a new IREE CMake invocation with its own set of
+# parameters.
+#
+# This function defines two custom target, `iree_prepare_${CONFIG_NAME}_dir`
+# and `iree_configure_${CONFIG_NAME}`, to drive the creation of a directory
+# for hosting the new IREE configuration and its corresponding `CMakeCache.txt`
+# file, respectively. Callers can then depend on either the file or the
+# target to make sure the configuration is invoked as a dependency.
```

Comment:
Can you expand on the difference between the two ways of depending on it? As written it sounds like you can depend on `iree_prepare_${CONFIG_NAME}_dir` to drive building the targets, which doesn't make sense (and I think isn't how it is). It doesn't seem there are any examples of depending on `iree_prepare_${CONFIG_NAME}_dir` right now other than `iree_configure_${CONFIG_NAME}`. That would just mean requiring that the directory has been created What's the use case where I want to depend on that dir existing, but not on the configuration actually happening outside of this function itself?

Reply:
Good point. `iree_prepare_${CONFIG_NAME}_dir` is internal. Removed it in the description to avoid confusion.